### PR TITLE
Port `LoginRequiredDialog` to an `AccountModalWidget`

### PR DESCRIPTION
**Pull request by erikjv**
And thereby not embed a complete `QDialog` in the settings window.

Fixes: #11592

### CODE REVIEW

**Comment by erikjv** [comment]
No time has been spend yet on the look, so the "new" look for now is:

Screenshot 2024-04-23 at 15 16 13

**Comment by erikjv** [comment]
How about this:

Screenshot 2024-04-24 at 12 14 27

File: src/gui/accountmodalwidget.h [hunk 1]
```diff
@@ -27,10 +28,19 @@ class AccountModalWidget : public QWidget
 public:
     AccountModalWidget(const QString &title, QWidget *widget, QWidget *parent);
 
+    enum DialogCode { Rejected, Accepted };
```
Comment (TheOneRing) [inline src/gui/accountmodalwidget.h:31]: enum class. and q_emum please.
Would Result be a better name?

**Comment by erikjv** [comment]
Wizard with basic auth

Screenshot 2024-04-24 at 13 33 45

File: src/gui/creds/httpcredentialsgui.cpp [hunk 2]
```diff
@@ -125,11 +123,13 @@ void HttpCredentialsGui::showDialog()
     auto *contentWidget = qobject_cast<BasicLoginWidget *>(dialog->contentWidget());
     contentWidget->forceUsername(user());
 
-    // in this case, we want to use the login button
-    dialog->addLogInButton();
-
-    connect(dialog, &LoginRequiredDialog::finished, ocApp()->gui()->settingsDialog(), [this, contentWidget](const int result) {
-        if (result == QDialog::Accepted) {
+    auto *modalWidget = new AccountModalWidget(tr("Login required"), dialog, ocApp()->gui()->settingsDialog());
+    modalWidget->setAttribute(Qt::WA_DeleteOnClose);
```
Comment (TheOneRing) [inline src/gui/creds/httpcredentialsgui.cpp:127]: Its no longer a dialog so this should not be needed.
https://github.com/owncloud/client/blob/815f10fe57736c3fe41bc0997a588cdf372a3d3a/src/gui/accountsettings.cpp#L589-L589

File: src/gui/creds/httpcredentialsgui.cpp [hunk 3]
```diff
@@ -186,8 +185,13 @@ void HttpCredentialsGui::restartOAuth()
             contentWidget->showRetryFrame();
         });
 
-        ocApp()->gui()->settingsDialog()->accountSettings(_account)->addModalLegacyDialog(
-            _loginRequiredDialog, AccountSettings::ModalWidgetSizePolicy::Minimum);
+        auto *modalWidget = new AccountModalWidget(tr("Login required"), _loginRequiredDialog, ocApp()->gui()->settingsDialog());
+        modalWidget->setAttribute(Qt::WA_DeleteOnClose);
```
Comment (TheOneRing) [inline src/gui/creds/httpcredentialsgui.cpp:189]:
```suggestion
```

**Review by TheOneRing — CHANGES_REQUESTED** [review]

File: src/gui/accountmodalwidget.h (hunk 1 above)
Comment (erikjv) [reply src/gui/accountmodalwidget.h:31]: done

File: src/gui/creds/httpcredentialsgui.cpp (hunk 3 above)
Comment (erikjv) [reply src/gui/creds/httpcredentialsgui.cpp]: done

**Review by TheOneRing — APPROVED** [review]
Is addModalLegacyDialog still used anywhere, if not could you open a follow-up and remove it?

**Comment by erikjv** [comment]
> Is addModalLegacyDialog still used anywhere, if not could you open a follow-up and remove it?

The share dialog and the folder wizard still use it.

**Pull request merged**